src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -608,19 +607,24 @@ private void setArgument(ArgumentDefinition argumentDefinition, List<String> val
             throw new CommandLineException("Argument '" + argumentDefinition.getNames() + "' cannot be specified more than once.");
         }
         if (argumentDefinition.isCollection) {
-            // if this is a collection then we only want to clear it once at the beginning, before we process
-            // any of the values
-            @SuppressWarnings("rawtypes")
-            final Collection c = (Collection) argumentDefinition.getFieldValue();
-            c.clear();
+            if (!parserOptions.contains(CommandLineParserOptions.APPEND_TO_COLLECTIONS)) {
+                // if this is a collection then we only want to clear it once at the beginning, before we process
+                // any of the values, unless we're in APPEND_TO_COLLECTIONS mode, in which case we leave the intial value
+                @SuppressWarnings("rawtypes")
+                final Collection c = (Collection) argumentDefinition.getFieldValue();
+                c.clear();
+            }
             values = expandListFile(values);
         }
 
         for (String stringValue: values) {
             final Object value;
             if (stringValue.equals(NULL_STRING)) {
-                if (argumentDefinition.isCollection && values.size() > 1) {
-                    // multiple values for a collection were passed, and at least one of them is null
+                if (argumentDefinition.isCollection &&
+                        values.size() > 1 &&
+                        !parserOptions.contains(CommandLineParserOptions.APPEND_TO_COLLECTIONS)) {
```

Comment:
Done. I've made it so that both modes accept and honor nulls at the beginning or mid-stream, and both issue a warning for the midstream case (while still honoring it, so that the previously specified values get clobbered).